apollo-runtime/src/main/java/com/apollographql/android/cache/TimeoutEvictionStrategy.java

```diff
@@ -0,0 +1,25 @@
+package com.apollographql.android.cache;
+
+import java.util.Date;
+
+import javax.annotation.Nonnull;
+
+import okhttp3.Response;
+import okhttp3.internal.http.HttpDate;
+
+public final class TimeoutEvictionStrategy implements ResponseCacheStore.EvictionStrategy {
+  private final long timeout;
+
+  public TimeoutEvictionStrategy(long timeout) {
+    this.timeout = timeout;
+  }
+
+  @Override public boolean isStale(@Nonnull Response response) {
+    String servedDateStr = response.header(HttpCacheInterceptor.CACHE_SERVED_DATE_HEADER);
+    if (servedDateStr == null) {
+      return true;
+    }
+    Date servedDate = HttpDate.parse(servedDateStr);
+    return servedDate == null || System.currentTimeMillis() - servedDate.getTime() > timeout;
```

Comment:
it shouldn't happen at all, as we always attach date to the header, but I thought if it happened then treat as always stale.